site/app/libraries/FileUtils.php

```diff
@@ -395,8 +395,56 @@ public static function getContentType($filename){
                 break;
             default:
                 $content_type = 'text/x-sh';
-                break;
+                break;$gradeable_ids = array_diff(scandir("/var/local/submitty/courses/$semester/$course/submissions/"), array('.', '..'));
         }
         return $content_type;
     }
+
+    /**
+     * This function returns a array consisting of all semesters, and within each semester, there will be courses for that semester, and
+     * within each course , there will be list of all gradeable ids for that course
+     * @return array $return
+     */
+    public static function getGradeablesFromPriorTerm(){
+        $return = array();
+
+        $filename = "/var/local/submitty/courses/gradeables_from_prior_terms.txt";
+        if (file_exists($filename)) {
+        
+          $file = fopen($filename, "r") or exit("Unable to open file!");
+
+          while(!feof($file)){
+            $line = fgets($file);
+            $line= trim($line," ");
+            $line= explode("/",$line);
+            $sem = $line[5];
+            $course = $line[6];
+            $gradeables= array();
+            while(!feof($file)){
+                $line = fgets($file);
+                if(trim(trim($line," "),"\n") == "") {
+                    break;
+                }
+                array_push($gradeables, trim(trim($line," "),"\n"));
+            }
+            $return[$sem][$course] = $gradeables;    
+            
+          }
+          fclose($file);
```

Comment:
What does the contents of this file look like?

Reply:
Contents will be like this:
/var/local/submitty/courses/f17/csci1100/submissions/:
hw01
hw02
hw03
hw04

/var/local/submitty/courses/f17/csci1200/submissions/:
final_exam
final_exam_notes
hw01
hw02

Reply:
The file is created using following command-
ls -1 /var/local/submitty/courses/*/*/submissions/ > /var/local/submitty/courses/gradeables_from_prior_terms.txt

This command will be later run by cronjob